src/opm/parser/eclipse/EclipseState/Grid/FieldProps.hpp

```diff
@@ -47,14 +48,14 @@ class FieldProps {
     template<typename T>
     struct FieldData {
         std::vector<T> data;
-        std::vector<bool> assigned;
+        std::vector<value::status> value_status;
 
 
         FieldData() = default;
 
         FieldData(std::size_t active_size) :
             data(std::vector<T>(active_size)),
-            assigned(std::vector<bool>(active_size, false))
+            value_status({active_size, value::status::uninitialized})
```

Comment:
Does this really do what you want?  The `initializer_list` constructor for `std::vector` assigns elements.  I'm actually surprised this compiles at all.  For safety you should write
```C++
value_status(active_size, value::status::uninitialized)
```
here

Reply:
Thanks - I have updated; I guess the original version might have created an additional copy and assigned that?